resources/sql/queries.sql

```diff
@@ -46,3 +46,60 @@ DELETE FROM services WHERE id = :id;
 -- :name delete-all-services! :! :n
 -- :doc deletes all service records
 DELETE FROM services;
+
+
+-- :name create-layout-table
+-- :command :execute
+-- :result :raw
+-- :doc Create widget layout table
+create table layout (
+  id                   VARCHAR(20) PRIMARY KEY,
```

Comment:
FYI, VARCHAR(30) is accepted by SQLite, but turns out it isn't idiomatic. The docs say the only "actual" supported types are: NULL, TEXT, INTEGER, REAL, and BLOB, so we are just making things harder on ourselves. And lengths are not supported or required.  See [https://www.sqlitetutorial.net/sqlite-data-types/](https://www.sqlitetutorial.net/sqlite-data-types/)

Reply:
Agree, sqlite doesnt seem to care at all.  I'll convert them all to TEXT and BLOB.  Should I convert the services table as well?

Reply:
Im not even sure the point of BLOB's either, I originally did that to try and prevent them from becoming strings, but even with a bunch of googling (and from running the code), sqlite just makes everything strings.

Reply:
yes, please convert everything (avoid confusion later) but TEST it well. I don't expect issues, but we've been burned before

Reply:
Done